lms/djangoapps/discussion_api/api.py

```diff
@@ -265,18 +280,24 @@ def get_thread_list(request, course_key, page, page_size, topic_id_list=None, te
 
     course = _get_course_or_404(course_key, request.user)
     context = get_context(course, request)
+
     query_params = {
+        "user_id": unicode(request.user.id),
         "group_id": (
             None if context["is_requester_privileged"] else
             get_cohort_id(request.user, course.id)
         ),
-        "sort_key": "date",
-        "sort_order": "desc",
         "page": page,
         "per_page": page_size,
         "text": text_search,
     }
+
+    cc_map = {"last_activity_at": "date", "comment_count": "comments", "vote_count": "votes"}
+    query_params["sort_key"] = cc_map.get(order_by) or "date"
```

Comment:
I would actually suggest that you raise an exception if an invalid sort_key is specified (preferably resulting in a 400 error)